src/aggregate.ts

```diff
@@ -157,4 +196,58 @@ class Count extends AggregateField {
   }
 }
 
+/**
+ * A PropertyAggregateField is a class that contains data that defines any
+ * aggregation that is performed on a property.
+ *
+ */
+abstract class PropertyAggregateField extends AggregateField {
+  property_?: string;
+  abstract operator: string;
+
+  /**
+   * Build a PropertyAggregateField object.
+   *
+   * @param {string} property
+   */
+  constructor(property: string) {
+    super();
+    this.property_ = property;
+  }
+
+  /**
+   * Gets the proto for the property aggregate field.
+   *
+   */
+  toProto(): any {
+    const aggregation = Object.assign(
+      {},
+      this.property_ ? {property: {name: this.property_}} : null
+    );
+    const aggregationObject: any = {};
+    aggregationObject[this.operator] = aggregation;
+    return Object.assign(
+      {operator: this.operator},
+      this.alias_ ? {alias: this.alias_} : null,
+      aggregationObject
```

Comment:
You can also drop the intermediate object and replace this line by
```js
{[this.operator]: aggregation}
```

Reply:
done